src/components/organisms/ChatSidebar/components/PrivateChats/PrivateChats.tsx

```diff
@@ -47,78 +56,71 @@ export const PrivateChats: React.FC<PrivateChatsProps> = ({ recipientId }) => {
-      onlineUsers
-        .filter((user) =>
-          user.partyName?.toLowerCase().includes(userSearchQuery.toLowerCase())
-        )
-        .map((user) => (
-          <OnlineUser
-            key={user.id}
-            user={user}
-            onClick={() => selectRecipientChat(user.id)}
-          />
-        )),
-    [onlineUsers, selectRecipientChat, userSearchQuery]
+      filteredUsers.slice(0, scrollPageNumber * PRIVATE_CHAT_NEXT_RENDER_SIZE),
+    [filteredUsers, scrollPageNumber]
   );
 
-  const numberOfSearchResults = renderedSearchResults.length;
+  const numberOfUsers = filteredUsers.length;
   const hasChatPreviews = renderedPrivateChatPreviews.length > 0;
-  const numberOfOtherOnlineUsers = renderedOnlineUsers.length;
 
   if (recipientId) {
     return <RecipientChat recipientId={recipientId} />;
   }
 
   return (
-    <div className="private-chats">
-      <InputField
-        containerClassName="private-chats__search"
-        placeholder="Search for online people"
-        value={userSearchQuery}
-        onChange={onInputChange}
-        iconStart={faSearch}
-        autoComplete="off"
-      />
-
-      {userSearchQuery ? (
-        <>
-          <p className="private-chats__title-text">
-            {numberOfSearchResults} search results
-          </p>
-
-          {renderedSearchResults}
-        </>
-      ) : (
-        <>
-          {hasChatPreviews && (
-            <div className="private-chats__previews">
-              {renderedPrivateChatPreviews}
-            </div>
-          )}
-
+    <div className="private-chats" id="private_chats_scrollable_div">
+      <InfiniteScroll
+        dataLength={renderedUsers.length}
+        hasMore={renderedUsers.length < filteredUsers.length}
+        next={showNextPage}
+        scrollableTarget="private_chats_scrollable_div"
+        loader={<Loading />}
+      >
+        <InputField
+          containerClassName="private-chats__search"
+          placeholder="Search for online people"
+          value={userSearchQuery}
+          onChange={onInputChange}
+          iconStart={faSearch}
+          autoComplete="off"
+        />
+
+        {userSearchQuery ? (
           <p className="private-chats__title-text">
-            {numberOfOtherOnlineUsers} other online people
+            {numberOfUsers} search results
           </p>
-
-          {renderedOnlineUsers}
-        </>
-      )}
+        ) : (
+          <>
+            {hasChatPreviews && (
+              <div className="private-chats__previews">
+                {renderedPrivateChatPreviews}
+              </div>
+            )}
+
+            <p className="private-chats__title-text">
+              {numberOfUsers} other online people
+            </p>
+          </>
+        )}
+
+        {renderedUsers.map((user) => (
+          <OnlineUser
+            key={user.id}
+            user={user}
+            onClick={() => selectRecipientChat(user.id)}
+          />
+        ))}
+      </InfiniteScroll>
```

Comment:
Why this was lifted out of `useMemo` and moved here?

Reply:
No special reason other than readability. I'll move it back

Reply:
Done

Reply:
@daniel-brown-sparkle  When replying with done (or equivalent) please copy paste the commit hash from where the change was made